src/parser/view_transform.c

```diff
@@ -13989,36 +14019,76 @@ mq_update_analytic_sort_spec_expr (PARSER_CONTEXT * parser, PT_NODE * spec, PT_N
 
 	  for (order = order_list; order; order = order->next)
 	    {
-	      PT_NODE *old_attr, *new_attr;
-	      int index;
+	      PT_NODE *old_select_node, *new_select_node;
+	      PT_NODE *from;
+	      int old_index, index;
 
 	      if (!PT_IS_VALUE_NODE (order->info.sort_spec.expr))
 		{
 		  continue;
 		}
 
 	      value = order->info.sort_spec.expr;
-	      /* retrieve the order-th node from the old_attrs */
-	      old_attr = pt_resolve_sort_spec_expr (parser, order, old_attrs);
-	      if (old_attr == NULL)
+	      old_index = value->info.value.data_value.i;
+
+	      if (old_select_list->type_enum != PT_TYPE_OBJECT)
 		{
-		  assert (false);
+		  old_index -= 1;
+		}
+
+	      old_select_node = pt_get_node_from_list (old_select_list, old_index);
+	      if (old_select_node == NULL)
+		{
+		  PT_ERRORmf (parser, order, MSGCAT_SET_PARSER_SEMANTIC, MSGCAT_SEMANTIC_SORT_SPEC_RANGE_ERR,
+			      old_index);
 		  return NULL;
 		}
 
-	      /* find the position of sort_spec expr in the new_attrs and update it with that position number */
-	      for (new_attr = new_attrs, index = 1; new_attr; new_attr = new_attr->next, index++)
+	      old_select_node = parser_copy_tree (parser, old_select_node);
+
+	      for (new_select_node = new_query->info.query.q.select.list, index = 1; new_select_node;
+		   new_select_node = new_select_node->next, index++)
 		{
-		  if (pt_str_compare (old_attr->info.name.original, new_attr->info.name.original, CASE_INSENSITIVE) ==
-		      0)
+		  if (new_select_node->type_enum == PT_TYPE_OBJECT)
 		    {
-		      value->info.value.data_value.i = index;
-		      order->info.sort_spec.pos_descr.pos_no = index;
+		      continue;
+		    }
 
-		      break;
+		  if ((old_select_node->node_type == PT_NAME || old_select_node->node_type == PT_DOT_)
+		      && (new_select_node->node_type == PT_NAME || new_select_node->node_type == PT_DOT_))
+		    {
+
+		      if (pt_check_path_eq (parser, old_select_node, new_select_node) == 0)
+			{
+			  /* name match */
+			  break;
+			}
+		    }
+		  else
+		    {
+		      /* brute method, compare printed trees */
+		      char *str_old = parser_print_tree (parser, old_select_node);
+		      char *str_new = parser_print_tree (parser, new_select_node);
+
+		      if (pt_str_compare (str_old, str_new, CASE_INSENSITIVE) == 0)
+			{
+			  /* match */
+			  break;
+			}
 		    }
 		}
```

Comment:
```suggestion
		  if (pt_check_path_eq (parser, old_select_node, new_select_node) == 0)
		    {
		      break;
		    }
		}		    
```
`pt_check_path_eq` 함수로 두 경우에 대한 비교가 모두 가능할 것 같습니다.
PT_NAME, PT_DOT인 경우와 그렇지 않은 경우로 나누게 비교해야 하는 경우 있을까요?

Reply:
리뷰해주신것처럼 pt_check_path_eq 함수는  PT_NAME, PT_DOT_인 경우 뿐만 아니라 다른 타입에도 적용할 수 있을 것 같습니다.

하지만, pt_check_path_eq 함수는  PT_NAME, PT_DOT_, PT_EXPR, PT_VALUE 타입의 노드만 비교할 수 있어, 다른 타입의 노드를 비교해야 할 필요가 있습니다.

select-list에 위치할 수 있는 노드의 타입은 다양하기 때문에, 더 많은 노드의 타입에 대해 비교해야 합니다.
예를 들어, PT_SELECT 타입의 스칼라 서브쿼리입니다.

Reply:
두 노드의 타입이 같고, 타입이 PT_NAME, PT_DOT_, PT_EXPR, PT_VALUE일 때는 `pt_check_path_eq` 함수로 비교하며,
그 외의 경우에 `pt_str_compare` 함수로 비교하도록 수정했습니다.

Reply:
PT_SELECT가 가능한 것은 처음 알았네요.
```
drop table if exists t1;
create table t1 (c1 int, c2 int);
select /*+ recompile */ row_number() over (partition by (select a.c1)), (select a.c1) from t1 a;

Query stmt:

select row_number() over (partition by 2), (select a.c1 from dual dual) from t1 a
```

기존 코드가 `pt_compare_sort_spec_expr` 함수의 구현 코드를 재사용했던 것이라면
`pt_compare_sort_spec_expr` 함수를 호출하는 코드로 변경하는 것이 유지보수에 좋을 것 같습니다.